File: elixir/raindrops/raindrops.exs
defmodule Raindrops do
  @sound_mapping %{ 3 => "Pling", 5 => "Plang", 7 => "Plong" }
  @empty_string ""
  def convert(input) do
    sounds = 
      @sound_mapping
      |> Enum.reduce(@empty_string, fn
        ({number, value}, output) when rem(input, number) == 0 -> output <> value 
        (_, output) -> output
      end)
    
    if sounds == @empty_string do
      to_string(input)
    else
      sounds
    end
  end
end

File: swift/raindrops/Raindrops.swift

class Raindrops {
    let sound_dict : [Int: String] = [3: "Pling", 5: "Plang", 7: "Plong"]

    var sounds : String

    init(_ raindrops : Int) {
        var output = "";
        for key in sound_dict.keys.filter({ x in raindrops % x == 0 }).sorted() {
            output += sound_dict[key]!
        }
        self.sounds = output != "" ? output : String(raindrops)
    }
}
